CMakeLists.txt

```diff
@@ -775,6 +775,7 @@ if(NOT MSVC)
   endif()
   if(CMAKE_COMPILER_IS_GNUCXX AND NOT (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0.0))
     string(APPEND CMAKE_CXX_FLAGS " -Wno-stringop-overflow")
+    string(APPEND CMAKE_CXX_FLAGS " -Wno-noexcept-type")
```

Comment:
Again, any explanation why this is needed?

Reply:
So gcc has a warning telling us that functions with noexcept annotation will have mangled name changed in C++17: https://stackoverflow.com/questions/46798456/handling-gccs-noexcept-type-warning

And since we have -Werror that becames an Error. I got it as a CI failure in previous PRs on files that I didnt modify which I don't know why my PR tickled.

I'll try removing it now since HEAD moved quite a bit to see if any failure shows up.

Reply:
removed.